packages/mdc-top-app-bar/index.js

```diff
@@ -0,0 +1,102 @@
+/**
+ *  @license
+ * Copyright 2018 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import MDCTopAppBarAdapter from './adapter';
+import MDCTopAppBarFoundation from './foundation';
+import MDCComponent from '@material/base/component';
+import {MDCRipple} from '@material/ripple/index';
+import {strings} from './constants';
+
+/**
+ * @extends {MDCComponent<!MDCTopAppBarFoundation>}
+ * @final
+ */
+class MDCTopAppBar extends MDCComponent {
+  /**
+   * @param {...?} args
+   */
+  constructor(...args) {
+    super(...args);
+    /** @private {?Element} */
+    this.navIcon_;
+    /** @type {?Array<MDCRipple>} */
+    this.iconRipples_;
+    /** @private {function(!Event)} */
+    this.navIconClick_;
+  }
+
+  initialize() {
+    this.navIcon_ = this.root_.querySelector(strings.MENU_ICON_SELECTOR);
+    this.navIconClick_ = this.navigationEvent_.bind(this);
+
+    if (this.navIcon_) {
+      this.navIcon_.addEventListener('click', this.navIconClick_);
```

Comment:
This much logic should not be in the component. This should be hooked up through the foundation with [de]registerInteractionHandler APIs.

Look at textfield/icon for example (not saying this has to be a subcomponent, but that shows an example of hooking up icon interaction + event emit through the foundation and that's almost the only thing that one does).